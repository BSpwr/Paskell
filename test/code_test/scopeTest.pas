program scopeTest;
var
   a, b, c: integer;

procedure scopeTest(a,b,c: integer);
begin
   a := 10;
   b := 20;
   c := a + b;

   writeln('In function scope');
   writeln('value of a = ', a , ' b =  ',  b, ' and c = ', c);
end;

begin
   a:= 100;
   b:= 200;
   c:= a + b;

   writeln('In global scope');
   writeln('value of a = ', a , ' b =  ',  b, ' and c = ', c);
   scopeTest(a, b, c);

  writeln('Back in global scope');
  writeln('value of a = ', a , ' b =  ',  b, ' and c = ', c);
end.