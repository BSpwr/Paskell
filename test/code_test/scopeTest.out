In global scope
value of a = 100 b =  200 and c = 300
In function scope
value of a = 10 b =  20 and c = 30
Back in global scope
value of a = 100 b =  200 and c = 300